src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -170,21 +170,23 @@ public bool HasResponseStarted
             get { return _requestProcessingStatus == RequestProcessingStatus.ResponseStarted; }
         }
 
-        protected FrameRequestHeaders FrameRequestHeaders => _frameHeaders.RequestHeaders;
+        protected FrameRequestHeaders FrameRequestHeaders { get; private set; }
+
+        protected FrameResponseHeaders FrameResponseHeaders { get; private set; }
 
         public void InitializeHeaders()
         {
-            if (_frameHeaders == null)
+            if (FrameRequestHeaders == null)
             {
-                _frameHeaders = new Headers(ServerOptions);
-                RequestHeaders = _frameHeaders.RequestHeaders;
-                ResponseHeaders = _frameHeaders.ResponseHeaders;
+                RequestHeaders = FrameRequestHeaders = new FrameRequestHeaders();
```

Comment:
Settable features. What a pain! It makes sense now though.

Is `RequestHeaders` even reset between requests? It doesn't look like it, and I think it needs to be. This seems to be a preexisting issue, but we might as well fix it now that we're looking at it.

It looks like `ResponseHeaders` is reset in `ProduceEnd` which works I guess.

@Tratcher Why is `IHttpResponseFeature.Headers` settable to begin with? It's not like Kestrel would send those headers to begin with.